### PR TITLE
fixed warning about size_t / int (..\apps\ca.c(2643): warning C4267: …

### PR DESCRIPTION
**Pull request by arminfuerst**
During the work on another branch I noticed that `app_malloc` uses `int` as parameter while `malloc` uses `size_t`. This leads to a warning in AppVeyor:
…'function': conversion from 'size_t' to 'int', possible loss of data)

When having tried to make this change in the other branch, it resulted in additional CI errors, so I am trying to submit this fix on it's own.

### CODE REVIEW

File: crypto/bio/bio_lib.c [hunk 1]
```diff
@@ -348,15 +348,15 @@ static int bio_write_intern(BIO *b, const void *data, size_t dlen,
     return ret;
 }
 
-int BIO_write(BIO *b, const void *data, int dlen)
+int BIO_write(BIO *b, const void *data, size_t dlen)
```
Comment (paulidale) [inline crypto/bio/bio_lib.c:351]: We can't change the signature of a public function.

Comment (arminfuerst) [reply crypto/bio/bio_lib.c:351]: > We can't change the signature of a public function.

Sorry, I wasn't aware BIO_write is a public function - I will revert it...

Comment (paulidale) [reply crypto/bio/bio_lib.c]: There are a number of other public functions you're modifying.  If a function is in `include/openssl/*.h`, it's considered public.

**Comment by arminfuerst** [comment]
The code in commit 48c65a6 is identical to commit 044f718 - any ideas why CI returns different results?

File: crypto/bio/bio_lib.c (hunk 1 above)
Comment (arminfuerst) [reply crypto/bio/bio_lib.c]: > There are a number of other public functions you're modifying. If a function is in `include/openssl/*.h`, it's considered public.

I already noticed and reverted all of those. My last commit is identical to the first commit which does not change any `include/openssl/` headers. What I don't understand: My first commit resulted in a green CI while the latest one was read. Any idea why there are different results to the same code?

**Comment by paulidale** [comment]
There is a known problem with the sanitizer builds which we've not tracked down yet.

**Comment by arminfuerst** [comment]
I see - I would submit the final state (only fixing the issue in apps.c). If I find solutions for other occurrences, I would make different PRs. Is it possible to submit the PR even if the last CI is red in this case? If not - how shall I continue?

**Comment by paulidale** [comment]
Keep going, we can merge with a red CI.  We just need to manually check if the CI is relevant before doing so.

**Review by mattcaswell — APPROVED** [review]
This looks fine to me so approving (although the PR still seems to be in a "Draft" state?)

**Comment by arminfuerst** [comment]
> This looks fine to me so approving (although the PR still seems to be in a "Draft" state?)

It was in Draft because I planned to do more and was unsure because of the red CI.

**Comment by arminfuerst** [comment]
> This PR is in a state where it requires action by @openssl/committers but the last update was 30 days ago

I was not aware I was supposed to take actions - what am I expected to do? I thought this PR just needs a second approve which cannot be done by myself?

**Comment by slontis** [comment]
> I thought this PR just needs a second approve which cannot be done by myself?

This is an automated message to inform committers that this PR has not been looked at for a while..

**Comment by arminfuerst** [comment]
> 
> 
> > I thought this PR just needs a second approve which cannot be done by myself?
> 
> This is an automated message to inform committers that this PR has not been looked at for a while..

But there is no action required from my side - I understood the message as if I was missing an action. I was aware that this PR was not changed for a long time but since it's not an important one, I didn't care.
As long as I didn't miss anything, I'm fine with it. I just wanted to ensure, the PR is not waiting for me.

**Review by slontis — COMMENTED** [review]
Hasnt this just moved the problem to somewhere else? i.e there are calls to this function that pass an int as the sz which could possibly be negative.

**Comment by arminfuerst** [comment]
> Hasnt this just moved the problem to somewhere else? i.e there are calls to this function that pass an int as the sz which could possibly be negative.

At least this was not my intention :)
In the first commit, I changed the relevant parameter from int to size_t, so there should be consequently size_t for one case. I made additional commits but I realized later I accidentally changed public headers, so I reverted the later commits and the final state should be identical to the first commit.
Of course, I might have made a mistake, I can check this later.

File: apps/include/apps.h [hunk 2]
```diff
@@ -203,7 +203,7 @@ typedef struct ca_db_st {
 } CA_DB;
 
 void app_bail_out(char *fmt, ...);
-void* app_malloc(int sz, const char *what);
+void* app_malloc(size_t sz, const char *what);
```
Comment (t8m) [inline apps/include/apps.h:206]: Could you please move adjust the spaces in the declaration to `void *app_malloc...`

File: apps/lib/apps.c [hunk 3]
```diff
@@ -628,7 +628,7 @@ void app_bail_out(char *fmt, ...)
     exit(1);
 }
 
-void* app_malloc(int sz, const char *what)
+void* app_malloc(size_t sz, const char *what)
```
Comment (t8m) [inline apps/lib/apps.c:631]: Same as above.

Comment (t8m) [reply apps/lib/apps.c:631]: Please also note that there is `Could not allocate %d bytes` format string which is clearly incorrect with size_t. Could you change it to `Could not allocate %zu bytes`...?

File: test/testutil/apps_mem.c [hunk 4]
```diff
@@ -11,7 +11,7 @@
 
 /* shim that avoids sucking in too much from apps/apps.c */
 
-void* app_malloc(int sz, const char *what)
+void* app_malloc(size_t sz, const char *what)
```
Comment (t8m) [inline test/testutil/apps_mem.c:14]: Same whitespace issue as above.

**Review by t8m — CHANGES_REQUESTED** [review]

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
@mattcaswell  still OK?

**Review by mattcaswell — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by t8m** [comment]
Merged to master. Thank you for the contribution!

**Pull request closed without merging**